### PR TITLE
Fixes #35

### LINKED ISSUE #35: Speaker website link
The speaker website link needs to prefix with http if the submitted value doesn't include it

### PR DESCRIPTION
**Pull request by philpursglove**
Checks the WebsiteUrl property for http/https and prepends it if it isn't there

### CODE REVIEW

**Comment by adrianbanks** [comment]
It looks like you need to rebase this onto the production branch.

**Comment by adrianbanks** [comment]
Also, I think we should not alter the user's entered value, but instead prepend the http/https bit when we output it to the screen.

**Comment by alastairs** [comment]
You could display it as a prefix to the input field like the @ symbol on the Twitter handle field, which would save people having to type it in. We could then add "http://" to the input, and run the resulting string through [`Uri.IsWellFormedUriString()`](http://msdn.microsoft.com/en-us/library/system.uri.iswellformeduristring.aspx) to validate it.

**Comment by philpursglove** [comment]
OK, will have another look at this today - not seen that Uri method before...

For the rebasing, is it 'git pull upstream/production SpeakerLinks --rebase'?

----- Original Message -----

**Comment by alastairs** [comment]
I normally do the following with the branch in question checked out:

git remote update --prune
git rebase upstream/production

I think yours will work if you have the branch checked out too, but be sure to check the log afterwards to make sure it looks ok!

**Comment by alastairs** [comment]
@philpursglove Has this now been rebased?

**Comment by philpursglove** [comment]
Have just finished getting quite cross at git and rebased it yes, just looking at putting the http: prefix into the view

**Comment by philpursglove** [comment]
Just pushed this but I'm not at all certain I've got all the git incantations correct to pick up all the production updates...

**Comment by alastairs** [comment]
Ummm. Looks like its got one of my commits in it somehow. Perhaps you missed a `git remote update` before rebasing the second time?

**Comment by philpursglove** [comment]
Bum - every time I think I'm getting the hang of this I find I'm actually not...

Looking back in my console window I did a 'rebase upstream/production', would that not have had the same effect as 'remote update'?

What's the fix, 'remote update' and push again?

----- Original Message -----

**Comment by ninjaferret** [comment]
I always do a _git fetch upstream_ first, this fetches all of the changes
from the main repo and onto the local and when I rebase onto
upstream/production it works (I think)

On 14 April 2013 20:35, Phil Pursglove notifications@github.com wrote:

> Bum - every time I think I'm getting the hang of this I find I'm actually
> not...
> 
> Looking back in my console window I did a 'rebase upstream/production',
> would that not have had the same effect as 'remote update'?
> 
> What's the fix, 'remote update' and push again?
> 
> ----- Original Message -----
> From: "Alastair Smith" notifications@github.com
> To: "dddeastanglia/DDDEastAnglia" DDDEastAnglia@noreply.github.com
> Cc: "Phil Pursglove" phil@philippursglove.com
> Sent: Sunday, 14 April, 2013 8:15:46 PM
> Subject: Re: [DDDEastAnglia] Fixes #35 (#51)
> 
> Ummm. Looks like its got one of my commits in it somehow. Perhaps you
> missed a git remote update before rebasing the second time?
> 
> —
> Reply to this email directly or view it on GitHub .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/dddeastanglia/DDDEastAnglia/pull/51#issuecomment-16357779
> .

## 

Ian Johnson

**Comment by philpursglove** [comment]
Can I rebase onto this commit and then redo my changes? Will that fix it?
image

File: DDDEastAnglia/Views/Speaker/Index.cshtml [hunk 1]
```diff
@@ -5,6 +5,11 @@
     ViewBag.Title = "Speakers";
 }
 
+@section featured
+{
+    @Html.Partial("_Banner")
+}
```
Comment (adrianbanks) [inline DDDEastAnglia/Views/Speaker/Index.cshtml]: When this is merged, this section will need to be removed, as the banner isn't done like this any more.

Comment (alastairs) [inline DDDEastAnglia/Views/Speaker/Index.cshtml]: Can the appropriate pull request be updated to remove this block then?

Comment (adrianbanks) [inline DDDEastAnglia/Views/Speaker/Index.cshtml]: You mean to pull this code out of this pull request?

Comment (alastairs) [inline DDDEastAnglia/Views/Speaker/Index.cshtml]: Yes. It's been added here, but it sounds like it's no longer appropriate to do so, so it should not be added any more.

**Comment by alastairs** [comment]
@philpursglove Your log looks correct, so I'm surprised that GitHub is including that commit of mine. Might be I have to rebase staging and you don't have to do anything at all (!).

**Comment by alastairs** [comment]
Yup, my bad, that should have fixed it now. Sorry!

File: DDDEastAnglia/Views/Speaker/Index.cshtml (hunk 1 above)
Comment (alastairs) [inline DDDEastAnglia/Views/Speaker/Index.cshtml]: @philpursglove Could you make this change so we can get this change merged?

**Comment by alastairs** [comment]
@philpursglove Please could you make the requested change so that we can get this merged? It will need rebasing onto the latest `production` branch, as well.

**Comment by philpursglove** [comment]
Sorry, I'd completely missed I needed to make another change to this! Not showing as automatically mergable though, and I thought I'd rebased :-(

**Comment by alastairs** [comment]
There are a couple of conflicts with @adrianbanks changes that are on staging. Adrian, please could you do this merge manually and resolve the conflicts? My mergetool isn't up to the job, it seems...

**Pull request merged**